### DISCUSSION TITLE
Utilizing Compressed Trajectory Formats (Bernstein Control Points)?

### DISCUSSION THREAD

**Discussion by csanchez9427** [USC-ACTLab/crazyswarm#752] [Q&A]
I have been using crazyswarm for basic flight patterns mainly using waypoints and uploadTrajectory functions. However, the lab I research at uses Bernstein Polynomials for all of their trajectory generation, and additionally they use MATLAB to generate these trajectories. 

The current method we use is to calculate the position, velocity, etc. for the entire trajectory based on the control points (the calculation is done in MATLAB), convert the data into the FullState message type in a python ROS node, and publish the data to the cmd_full_state topics at a specific rate (usually 10-30 Hz). However, we are planning to expand to a very large number of crazyflies and are worried about bandwidth. 

We noticed the crazyflie-firmware includes a "compressed" trajectory format, which seems to be able to take just control points for its flight path. Is there anyway to utilize this format in the crazyswarm platform? Would there be anyway for us to just send control points to the drones every couple of seconds instead of sending the desired position, velocity, etc. multiple times a second?

**Comment by whoenig** [comment]
The compressed trajectory has not been implemented in the Crazyswarm (nor crazyflie_cpp), but it wouldn't be hard to do so. We only maintain Crazyswarm2 now, since ROS1 will be discontinued soon. Crazyswarm2 does not have this feature yet, either.

**Reply by csanchez9427** [reply]
Thanks for the reply! 
Is there anywhere this function is readily available, or where I could learn to implement this function?

**Reply by whoenig** [reply]
I would guess in cflib is such support, but you would need to check with Bitcraze. To implement yourself, the easiest is to find a similar example for what you want to achieve (the uncompressed trajectory I think) and then adjust it to your use-case.
